test/functions.js

```diff
@@ -694,6 +694,9 @@
       assert.deepEqual(_.toArray(cb(1, 2, 3, 4, 5, 6, 7, 8, 9, 10)), _.range(1, 11));
     });
 
+    var deepProperty = _.iteratee(['a', 'b']);
+    assert.equal(deepProperty({a: {b: 2}}), 2, 'treats an array as a deep property accessor');
+
```

Comment:
> In lodash you use arrays as shorthand for a type of predicate which can already be expressed via an object (`_.matches`)?

Yep. The `_.matches` shorthand is a bit heavier as it supports multiple properties.
You can think of `_.matches` implemented by way of `_.matchesProperty`. The `_.matches` method handles a collection of source properties to compare.

> I'm guessing from the phrase "fast path" and the perf graph you included that this somehow offers a performance gain?

Yep. It's a common enough case that a fast path was created.

> To me, deep property matching seems much more useful than a performance increase which users could presumably get in the places where performance was a concern by simply using `_.matchesProperty()` or their own optimized function directly.

In lodash `matches` and `matchesProperty` already perform partial deep matches of source objects:
```js
_.filter(objects, { 'a': { 'b': { 'c': 1 }, 'd': 2 } });
```
In Lodash the common case for deep property paths is strings in with arrays as a fallback for edge cases which is where `_.property(array)` would be used.
```js
_.filter(objects, 'a.b.c');
```